src/GafferScene/LightFilter.cpp

```diff
@@ -137,11 +149,17 @@ IECore::ConstObjectPtr LightFilter::computeSource( const Context *context ) cons
 void LightFilter::hashAttributes( const SceneNode::ScenePath &path, const Gaffer::Context *context, const ScenePlug *parent, IECore::MurmurHash &h ) const
 {
 	h.append( shaderPlug()->attributesHash() );
+	filteredLightsPlug()->hash( h );
```

Comment:
We're missing a call to the base class hash in this function. It's a bit complicated because ObjectSource has a specialised implementation that isn't what we want, so the right thing to do is call through to SceneNode. Could you add this at the top of this function?

```
// We must call the base class before appending to the hash, but our direct
// base class (ObjectSource) is set up with a hardcoded hash suitable only
// for outputting empty attributes. Call directly to our SceneNode ancestor instead.
SceneNode::hashAttributes( path, context, parent );
```

Reply:
Done here: 237d6e0

I'll assume the missing hash in the arguments was a typo and not some intentional thing I just don't understand :)